code/modules/mining/lavaland/weather.dm

```diff
@@ -0,0 +1,89 @@
+var/global/activeWeather = FALSE
+
+/datum/weather
+	var/name = "storm"
+	var/start_up_time = 300 //30 seconds
+	var/start_up_message = "The wind begins to pick up."
+	var/duration = 1200 //2 minutes
+	var/duration_message = "A storm has started!"
+	var/wind_down = 300 // 30 seconds
+	var/wind_down_message = "The storm is passing."
+
+	var/target_z = 1
+	var/exclude_walls = TRUE
+	var/area_type = /area/space
+
+	var/start_up_overlay = "lava"
+	var/duration_overlay = "lava"
+	var/purely_aesthetic = FALSE //If we just want gentle rain that doesn't hurt people
+
+
+
+/datum/weather/proc/weather_start_up()
+	if(activeWeather)
+		return
+	activeWeather = TRUE
```

Comment:
Should make this `activeWeather = src`; could be helpful in the future and will achieve the same result